lib/faraday/utils.rb

```diff
@@ -95,7 +95,15 @@ def to_hash() ::Hash.new.update(self) end
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Find the last set of response headers.
+        last_response = headers.reverse.take(
+          headers.reverse.find_index { |x| x.match(/^HTTP\//) } + 1
+        ).reverse
```

Comment:
The above is extremely inefficient.
Please change it to:

```ruby
start_index = headers.rindex { |x| x.match(/^HTTP\//) }
last_response = headers.slice(start_index, headers.size)
```

Or happy to review an even better proposal 😄